test/utils/ExternalMulticallSimulator.sol.ts

```diff
@@ -0,0 +1,359 @@
+import type { HardhatEthersSigner } from '@nomicfoundation/hardhat-ethers/signers';
+import * as helpers from '@nomicfoundation/hardhat-network-helpers';
+import { expect } from 'chai';
+import type { AddressLike, BigNumberish, BytesLike, Interface, VoidSigner } from 'ethers';
+import { ethers } from 'hardhat';
+
+import { type DapiProxyWithOev, DapiProxyWithOev__factory } from '../../src/index';
+import * as testUtils from '../test-utils';
+
+interface Call {
+  target: AddressLike;
+  data: BytesLike;
+}
+
+async function getDapiTransmutationCalldata(
+  preTransmutationCalls: Call[],
+  postTransmutationCalls: Call[],
+  api3ServerV1Address: AddressLike,
+  api3ServerV1Interface: Interface,
+  dapiName: BytesLike,
+  value: BigNumberish,
+  externalMulticallSimulatorInterface: Interface
+) {
+  // Generating a private key may be a bit too compute-intensive. We can hardcode a mock one instead.
+  const MOCK_AIRNODE_PRIVATE_KEY = '0x0fbcf3c01c9bcde58a6efa722b8d9019043dfaf5cdf557693442732e24b9f5ab';
+  const airnode = new ethers.BaseWallet(new ethers.SigningKey(MOCK_AIRNODE_PRIVATE_KEY));
+  // We want to use a Beacon ID that no one else has used to avoid griefing. Randomly generating the
+  // template ID would solve that.
+  const templateId = testUtils.generateRandomBytes32();
+  const timestamp = Math.floor(Date.now() / 1000);
+  const data = ethers.AbiCoder.defaultAbiCoder().encode(['int256'], [value]);
+  const signature = await testUtils.signData(airnode, templateId, timestamp, data);
+  const beaconId = testUtils.deriveBeaconId(airnode.address, templateId);
+  const transmutationCalls = [
+    {
+      target: api3ServerV1Address,
+      data: api3ServerV1Interface.encodeFunctionData('setDapiName', [dapiName, beaconId]),
```

Comment:
Btw. this works great for dAPI proxies, but this won't work for data feed ones. I don't see how we could make that work though (so just thinking out loud).

Reply:
Since this is a DapiTransmuter, I left that out of scope. There's no way of making it work with data feeds anyway.